src/panels/ExoplanetsPanel/ExoplanetsPanel.tsx

```diff
@@ -48,37 +46,32 @@ export function ExoplanetsPanel() {
     owner!.tags.includes('exoplanet_system')
   );
 
-  const justAdded = loadingAdded.filter(
-    (name) => addedSystems.findIndex((s) => s && s.name.includes(name)) !== -1
-  );
-  const justRemoved = loadingRemoved.filter(
-    (name) => addedSystems.findIndex((s) => s && s.name.includes(name)) === -1
-  );
-  if (justAdded.length > 0) {
-    const newAdded = loadingAdded.filter((e) => !justAdded.includes(e));
-    setLoadingAdded(newAdded);
-  }
-  if (justRemoved.length > 0) {
-    const newRemoved = loadingRemoved.filter((e) => !justRemoved.includes(e));
-    setLoadingRemoved(newRemoved);
+  function isAdded(starName: string) {
+    // Replace all spaces with underscores
+    return (
+      addedSystems.findIndex(
+        (owner) => owner?.identifier === name2Identifier(starName)
+      ) !== -1
+    );
   }
 
-  function removeSystem(starName: string, identifier: string) {
-    // In case we happen to be focused on the removed system star, reset the focus
-    // @TODO (emmbr, 2024-11-29): This will still not check if any of the child nodes are
-    // removed... should be fixed, by setting the anchor/aim property correctly on the
-    // OpenSpace side instead
-    if (anchor === identifier || aim === identifier) {
-      dispatch(setPropertyValue({ uri: NavigationAnchorKey, value: 'Sun' }));
-      dispatch(setPropertyValue({ uri: NavigationAimKey, value: '' }));
-    }
-    luaApi?.exoplanets.removeExoplanetSystem(starName);
-    setLoadingRemoved([...loadingRemoved, starName]);
+  function name2Identifier(starName: string) {
```

Comment:
We should use the Lua function for this instead: 
```
  openspace.makeIdentifier(string)
```
This is the one that's being used by the exoplanet module for the star name :) And it does a bit more than just replacing whitespace with underscores